ProcessExitTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\RunInSeparateProcess;
+use PHPUnit\Framework\TestCase;
+
+final class ProcessExitTest extends TestCase
+{
+    #[RunInSeparateProcess]
+    #[DataProvider("provideExitCodes")]
+    public function testOne(?int $expectedExit, int $actualExitCode): void
+    {
+        if ($expectedExit !== null) {
+            $this->expectProcessExit($expectedExit);
```

Comment:
we should add a PHPUnit error/warning when `expectProcessExit` is called in a test, which does not use process-isolation